pkg/controllers/cspc-controller/util/cspc_util_test.go

```diff
@@ -0,0 +1,171 @@
+package util
+
+import (
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+
+	"testing"
+)
+
+var (
+	condPoolManagerAvailable = func() cstor.CStorPoolClusterCondition {
+		return cstor.CStorPoolClusterCondition{
+			Type:   PoolManagerAvailable,
+			Status: corev1.ConditionTrue,
+			Reason: "AwesomeCSPCController",
+		}
+	}
+
+	condPoolManagerImaginary = func() cstor.CStorPoolClusterCondition {
+		return cstor.CStorPoolClusterCondition{
+			Type:   "Imaginary",
+			Status: corev1.ConditionTrue,
+			Reason: "AwesomeCSPCController",
+		}
+	}
+
+	condPoolManagerImaginary1 = func() cstor.CStorPoolClusterCondition {
+		return cstor.CStorPoolClusterCondition{
+			Type:   "Imaginary",
+			Status: corev1.ConditionFalse,
+			Reason: "ForSomeReason",
+		}
+	}
+
+	status = func() *cstor.CStorPoolClusterStatus {
+		return &cstor.CStorPoolClusterStatus{
+			Conditions: []cstor.CStorPoolClusterCondition{condPoolManagerImaginary(), condPoolManagerAvailable()},
+		}
+	}
+)
+
+func TestGetCondition(t *testing.T) {
+	exampleStatus := status()
+
+	tests := []struct {
+		name string
+
+		status   cstor.CStorPoolClusterStatus
+		condType cstor.CSPCConditionType
+
+		expected bool
+	}{
+		{
+			name: "condition exists",
+
+			status:   *exampleStatus,
+			condType: PoolManagerAvailable,
+
+			expected: true,
+		},
+		{
+			name: "condition does not exist",
+
+			status:   *exampleStatus,
+			condType: "testCondition",
+
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cond := GetCSPCCondition(test.status, test.condType)
+			exists := cond != nil
+			if exists != test.expected {
+				t.Errorf("%s: expected condition to exist: %t, got: %t", test.name, test.expected, exists)
+			}
+		})
+	}
+}
+
+func TestSetCondition(t *testing.T) {
+	tests := []struct {
+		name string
+
```

Comment:
remove extra lines , i think go fmt run required in all these files .

Reply:
Go fmt does not remove extra lines. I think I need to remove it.

Reply:
thanks